### ISSUE TITLE
groupBy example vs HttpClient.get

### ISSUE THREAD

**Issue by IvanKostyuk** [ReactiveX/rxjs#5459]
### Documentation Related To Component:

/api/operators/groupBy

### Please check those that apply

- [ ] typo
- [ ] documentation doesn't exist
- [ ] documentation needs clarification
- [v] error(s) in example
- [v ] needs example

### Description Of The Issue
Using version 6.5.4 with Angular 9.1.7
I implemented the example, everything works as expected after addition of `toArray(),`, without this only last record can be seen. I implemented code from the example on my application. getting this
`[ [ "undefined", null ] ]`
when doing this:

HTML part `<div>{{plan | json}}</div>`

```js
 plan: Object;

  getData(): void {
    this.heroService.getPlan()
    .pipe(
      groupBy(p => p.grade, p => p.subject),
      mergeMap(group$ =>
        group$.pipe(reduce((acc, cur) => [...acc, cur], [`${group$.key}`]))
      ),
      toArray(),
   )
   .subscribe(p => this.plan = p);
  }
```

```js
interface PlanRecord  {
    grade: number;
    subject: string;
  }
  getPlan(): Observable<PlanRecord> {
    const url = `${this.planUrl}`;
    return this.http.get<PlanRecord>(url).pipe(
      tap(_ => this.log(`fetched student's plan`)),
      catchError(this.handleError<PlanRecord>(`getPlan`))
    );
  }
```

input data
```js
const plan = [
    { grade: 8, subject: 'English'},
    { grade: 8, subject: 'Spanish'},
    { grade: 8, subject: 'Chemistry'},
    { grade: 8, subject: 'Algebra'},
    { grade: 9, subject: 'Engineering'}
  ];
```

These code works successfully
```js
  getPlan(): void {
    this.heroService.getPlan()
      .subscribe(plan => this.planArray = plan);
  }
```

Example with real API on component level will really help to understand it better.

**Comment by IvanKostyuk** [comment]
Small addition. When I change
```js
groupBy(p => p.grade, p => p.subject),
```
to 
```js
groupBy(p => p.grade, p => p),
```
then I see results grouped by "undefined" rather then by "8" and "9", but I see all the records in that group.

**Comment by Andrei0872** [comment]
As far as I can see, if `heroService.getPlan()` returns that `input`, it means that `groupBy` will be given an array of objects, instead of an object.

So what you could do is to add `mergeAll()` before `groupBy` in order to get each object individually.

**Comment by IvanKostyuk** [comment]
Yes, Andrei, you've got it right. Like most of simple services getPlan() returns an array, not stream.
I started to learn TypeScript, Angular and RxJs just 4 days ago.
`mergeAll()` solved one problem, but introduced another one in the same example.

Thank you, Andrei, for showing me right direction.

Another problem happened, when I moved to more real results
```ts
      mergeMap(group$ =>
        group$.pipe(reduce((acc, cur) => [...acc, cur], [`${group$.key}`]))
      ),
      map(arr => ({ id: parseInt(arr[0], 10), values: arr.slice(1) }))
```
the problem is due to _union type_ of `arr` - `(string | Plan)[]`. `acc` is `string[]` and `cur` is `Plan`, therefore `arr[0]` has to be read as `string` in order to have proper assignment.

As I'm new in this stack of technologies, I think it will benefit to all, to mention in the documentation to pay attention to _higher-order Observables_ when developer deals with real or fake API calls instead of embedded JSONs (printing to console is not a goal).

I'm checking how to update documentation (I hope update of `src/internal/operators/groupBy.ts` will be sufficient, but I'm not sure).

Any suggestion on better wording?
